### ISSUE TITLE
HRSC KLD data reproduction

### ISSUE THREAD

**Issue by Kn-Oh** [open-mmlab/mmrotate#817]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmrotate/issues) and [Discussions](https://github.com/open-mmlab/mmrotate/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmrotate.readthedocs.io/en/1.x/notes/4_faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (master)](https://github.com/open-mmlab/mmrotate) or [latest version (1.x)](https://github.com/open-mmlab/mmrotate/tree/dev-1.x).

### Task

I'm using the official example scripts/configs for the officially supported tasks/models/datasets.

### Branch

master branch https://github.com/open-mmlab/mmrotate

### Environment

(mmroate) ➜  mmrotate git:(qd) ✗ python mmrotate/utils/collect_env.py
sys.platform: linux
Python: 3.8.15 (default, Nov 24 2022, 15:19:38) [GCC 11.2.0]
CUDA available: True
numpy_random_seed: 2147483648
GPU 0: Tesla V100-SXM2-32GB
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 11.4, V11.4.152
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.10.0+cu111
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.2.3 (Git Hash 7336ca9f055cf1bfa13efb658fe15dc9b41f0740)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX512
  - CUDA Runtime 11.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.1, CUDNN_VERSION=8.0.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.10.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.11.0+cu111
OpenCV: 4.6.0
MMEngine: 0.4.0
MMRotate: 1.0.0rc0+3db9cc0

### Reproduces the problem - code sample

dataset_type = 'HRSCDataset'
data_root = '/root/autodl-tmp/HRSC/HRSC2016/'
img_norm_cfg = dict(
mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
dict(type='LoadImageFromFile'),
dict(type='LoadAnnotations', with_bbox=True),
dict(type='RResize', img_scale=(800, 512)),
dict(
type='RRandomFlip',
flip_ratio=[0.25, 0.25, 0.25],
direction=['horizontal', 'vertical', 'diagonal'],
version='le90'),
dict(
type='PolyRandomRotate',
rotate_ratio=0.5,
angles_range=180,
auto_bound=False,
rect_classes=[9, 11],
version='le90'),
dict(
type='Normalize',
mean=[123.675, 116.28, 103.53],
std=[58.395, 57.12, 57.375],
to_rgb=True),
dict(type='Pad', size_divisor=32),
dict(type='DefaultFormatBundle'),
dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
]
test_pipeline = [
dict(type='LoadImageFromFile'),
dict(
type='MultiScaleFlipAug',
img_scale=(800, 512),
flip=False,
transforms=[
dict(type='RResize'),
dict(
type='Normalize',
mean=[123.675, 116.28, 103.53],
std=[58.395, 57.12, 57.375],
to_rgb=True),
dict(type='Pad', size_divisor=32),
dict(type='DefaultFormatBundle'),
dict(type='Collect', keys=['img'])
])
]
data = dict(
samples_per_gpu=2,
workers_per_gpu=2,
train=dict(
type='HRSCDataset',
classwise=False,
ann_file='/root/autodl-tmp/HRSC/HRSC2016/ImageSets/trainval.txt',
ann_subdir='/root/autodl-tmp/HRSC/HRSC2016/FullDataSet/Annotations/',
img_subdir='/root/autodl-tmp/HRSC/HRSC2016/FullDataSet/AllImages/',
img_prefix='/root/autodl-tmp/HRSC/HRSC2016/FullDataSet/AllImages/',
pipeline=[
dict(type='LoadImageFromFile'),
dict(type='LoadAnnotations', with_bbox=True),
dict(type='RResize', img_scale=(800, 512)),
dict(
type='RRandomFlip',
flip_ratio=[0.25, 0.25, 0.25],
direction=['horizontal', 'vertical', 'diagonal'],
version='le90'),
dict(
type='PolyRandomRotate',
rotate_ratio=0.5,
angles_range=180,
auto_bound=False,
rect_classes=[9, 11],
version='le90'),
dict(
type='Normalize',
mean=[123.675, 116.28, 103.53],
std=[58.395, 57.12, 57.375],
to_rgb=True),
dict(type='Pad', size_divisor=32),
dict(type='DefaultFormatBundle'),
dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
],
version='le90'),
val=dict(
type='HRSCDataset',
classwise=False,
ann_file='/root/autodl-tmp/HRSC/HRSC2016/ImageSets/test.txt',
ann_subdir='/root/autodl-tmp/HRSC/HRSC2016/FullDataSet/Annotations/',
img_subdir='/root/autodl-tmp/HRSC/HRSC2016/FullDataSet/AllImages/',
img_prefix='/root/autodl-tmp/HRSC/HRSC2016/FullDataSet/AllImages/',
pipeline=[
dict(type='LoadImageFromFile'),
dict(
type='MultiScaleFlipAug',
img_scale=(800, 512),
flip=False,
transforms=[
dict(type='RResize'),
dict(
type='Normalize',
mean=[123.675, 116.28, 103.53],
std=[58.395, 57.12, 57.375],
to_rgb=True),
dict(type='Pad', size_divisor=32),
dict(type='DefaultFormatBundle'),
dict(type='Collect', keys=['img'])
])
],
version='le90'),
test=dict(
type='HRSCDataset',
classwise=False,
ann_file='/root/autodl-tmp/HRSC/HRSC2016/ImageSets/test.txt',
ann_subdir='/root/autodl-tmp/HRSC/HRSC2016/FullDataSet/Annotations/',
img_subdir='/root/autodl-tmp/HRSC/HRSC2016/FullDataSet/AllImages/',
img_prefix='/root/autodl-tmp/HRSC/HRSC2016/FullDataSet/AllImages/',
pipeline=[
dict(type='LoadImageFromFile'),
dict(
type='MultiScaleFlipAug',
img_scale=(800, 512),
flip=False,
transforms=[
dict(type='RResize'),
dict(
type='Normalize',
mean=[123.675, 116.28, 103.53],
std=[58.395, 57.12, 57.375],
to_rgb=True),
dict(type='Pad', size_divisor=32),
dict(type='DefaultFormatBundle'),
dict(type='Collect', keys=['img'])
])
],
version='le90'))
evaluation = dict(interval=10, metric='mAP')
optimizer = dict(type='SGD', lr=0.01, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=dict(max_norm=35, norm_type=2))
lr_config = dict(
policy='step',
warmup='linear',
warmup_iters=500,
warmup_ratio=0.3333333333333333,
step=[48, 66])
runner = dict(type='EpochBasedRunner', max_epochs=100)
checkpoint_config = dict(interval=36)
log_config = dict(interval=50, hooks=[dict(type='TextLoggerHook')])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
angle_version = 'le90'
model = dict(
type='RotatedRetinaNet',
backbone=dict(
type='ResNet',
depth=50,
num_stages=4,
out_indices=(0, 1, 2, 3),
frozen_stages=1,
zero_init_residual=False,
norm_cfg=dict(type='BN', requires_grad=True),
norm_eval=True,
style='pytorch',
init_cfg=dict(type='Pretrained', checkpoint='torchvision://resnet50')),
neck=dict(
type='FPN',
in_channels=[256, 512, 1024, 2048],
out_channels=256,
start_level=1,
add_extra_convs='on_input',
num_outs=5),
bbox_head=dict(
type='RotatedRetinaHead',
num_classes=1,
in_channels=256,
stacked_convs=4,
feat_channels=256,
assign_by_circumhbbox=None,
anchor_generator=dict(
type='RotatedAnchorGenerator',
octave_base_scale=4,
scales_per_octave=3,
ratios=[1.0, 0.5, 2.0],
strides=[8, 16, 32, 64, 128]),
bbox_coder=dict(
type='DeltaXYWHAOBBoxCoder',
angle_range='le90',
norm_factor=None,
edge_swap=True,
proj_xy=True,
target_means=(0.0, 0.0, 0.0, 0.0, 0.0),
target_stds=(1.0, 1.0, 1.0, 1.0, 1.0)),
loss_cls=dict(
type='FocalLoss',
use_sigmoid=True,
gamma=2.0,
alpha=0.25,
loss_weight=1.0),
loss_bbox=dict(
type='GDLoss_v1',
loss_type='kld',
fun='log1p',
tau=1,
loss_weight=1.0),
reg_decoded_bbox=True),
train_cfg=dict(
assigner=dict(
type='MaxIoUAssigner',
pos_iou_thr=0.5,
neg_iou_thr=0.4,
min_pos_iou=0,
ignore_iof_thr=-1,
iou_calculator=dict(type='RBboxOverlaps2D')),
allowed_border=-1,
pos_weight=-1,
debug=False),
test_cfg=dict(
nms_pre=2000,
min_bbox_size=0,
score_thr=0.05,
nms=dict(iou_thr=0.1),
max_per_img=2000))
work_dir = '/root/autodl-tmp/hrsc_kld'
auto_resume = False
gpu_ids = range(0, 4)

### Reproduces the problem - command or script

cd mmrotate
cd tools
./dist_train.sh

### Reproduces the problem - error message

Epoch(val) [100][114]	mAP: 0.4531, AP50: 0.8550, AP55: 0.8490, AP60: 0.8210, AP65: 0.7230, AP70: 0.6000, AP75: 0.3960, AP80: 0.2180, AP85: 0.0530, AP90: 0.0150, AP95: 0.0010
+-------+------+------+--------+-------+
| class | gts  | dets | recall | ap    |
+-------+------+------+--------+-------+
| ship  | 1228 | 1889 | 0.967  | 0.855 |
+-------+------+------+--------+-------+
| mAP   |      |      |        | 0.855 |
+-------+------+------+--------+-------+
2023-04-12 11:43:39,065 - mmrotate - INFO - 
+-------+------+------+--------+-------+
| class | gts  | dets | recall | ap    |
+-------+------+------+--------+-------+
| ship  | 1228 | 1889 | 0.954  | 0.849 |
+-------+------+------+--------+-------+
| mAP   |      |      |        | 0.849 |
+-------+------+------+--------+-------+
2023-04-12 11:43:42,773 - mmrotate - INFO - 
+-------+------+------+--------+-------+
| class | gts  | dets | recall | ap    |
+-------+------+------+--------+-------+
| ship  | 1228 | 1889 | 0.910  | 0.821 |
+-------+------+------+--------+-------+
| mAP   |      |      |        | 0.821 |
+-------+------+------+--------+-------+
2023-04-12 11:43:46,586 - mmrotate - INFO - 
+-------+------+------+--------+-------+
| class | gts  | dets | recall | ap    |
+-------+------+------+--------+-------+
| ship  | 1228 | 1889 | 0.836  | 0.723 |
+-------+------+------+--------+-------+
| mAP   |      |      |        | 0.723 |
+-------+------+------+--------+-------+
2023-04-12 11:43:50,243 - mmrotate - INFO - 
+-------+------+------+--------+-------+
| class | gts  | dets | recall | ap    |
+-------+------+------+--------+-------+
| ship  | 1228 | 1889 | 0.724  | 0.600 |
+-------+------+------+--------+-------+
| mAP   |      |      |        | 0.600 |
+-------+------+------+--------+-------+
2023-04-12 11:43:53,966 - mmrotate - INFO - 
+-------+------+------+--------+-------+
| class | gts  | dets | recall | ap    |
+-------+------+------+--------+-------+
| ship  | 1228 | 1889 | 0.567  | 0.396 |
+-------+------+------+--------+-------+
| mAP   |      |      |        | 0.396 |
+-------+------+------+--------+-------+
2023-04-12 11:43:57,629 - mmrotate - INFO - 
+-------+------+------+--------+-------+
| class | gts  | dets | recall | ap    |
+-------+------+------+--------+-------+
| ship  | 1228 | 1889 | 0.354  | 0.218 |
+-------+------+------+--------+-------+
| mAP   |      |      |        | 0.218 |
+-------+------+------+--------+-------+
2023-04-12 11:44:01,451 - mmrotate - INFO - 
+-------+------+------+--------+-------+
| class | gts  | dets | recall | ap    |
+-------+------+------+--------+-------+
| ship  | 1228 | 1889 | 0.150  | 0.053 |
+-------+------+------+--------+-------+
| mAP   |      |      |        | 0.053 |
+-------+------+------+--------+-------+
2023-04-12 11:44:05,155 - mmrotate - INFO - 
+-------+------+------+--------+-------+
| class | gts  | dets | recall | ap    |
+-------+------+------+--------+-------+
| ship  | 1228 | 1889 | 0.024  | 0.015 |
+-------+------+------+--------+-------+
| mAP   |      |      |        | 0.015 |
+-------+------+------+--------+-------+
2023-04-12 11:44:08,842 - mmrotate - INFO - 
+-------+------+------+--------+-------+
| class | gts  | dets | recall | ap    |
+-------+------+------+--------+-------+
| ship  | 1228 | 1889 | 0.002  | 0.001 |
+-------+------+------+--------+-------+
| mAP   |      |      |        | 0.001 |
+-------+------+------+--------+-------+

### Additional information

This configuration is the configuration of kld about hrsc and rotated_retinanet detector released by Dr. Yang Xue on github. The only difference is that there is one more path, and deleting this path will cause an error. The results of this configuration are far from the data in the paper. The kld configuration data from the toolbox is similar to this data. I don't know what is wrong, hope you can help me, thank you

**Comment by zytx121** [comment]
Hi @Kn-Oh, I noticed that you are using 4 GPUs. However, all of our KLD experiments only used one GPU.

**Comment by Kn-Oh** [comment]
> Hi @Kn-Oh, I noticed that you set `gpu_ids = range(0, 4)`. However, all of our KLD experiments only used one GPU.

Yes, I ran it with 4 GPUs, and I also adjusted the learning rate to 0.01, but does this have such a big impact on the accuracy? And when I was running, the accuracy was only around 80 in about 50 epochs. It is far from the precision of more than 80 in 12 epochs introduced in the previous HRSC recurring issue. Since I use 4 GPUs to ensure that the map and the highest AP can reach the accuracy of the paper, I increased the epoch to ensure the number of training rounds, but it still cannot reach the accuracy of the paper. Hope you can help me out, thanks

**Comment by exesit** [comment]
我也存在这种问题
